### PR TITLE
User verification message

### PR DESCRIPTION
**Pull request by lupyana**
# What is the Purpose?
if unverified user tries to login notify them to verify the account first

## What was the approach?
Show an error message when user tries to login

## Are there any concerns to addressed further before or after merging this PR?
My main concern is localization of the error message, Im not really sure how to handle this @odenypeter @michaelbukachi  please advise

## Mentions?
@andrewtpham 
@odenypeter 

## Issue(s) affected?
see #29

### CODE REVIEW

File: src/utils/request.js [hunk 1]
```diff
@@ -46,6 +46,12 @@ service.interceptors.response.use(
     } else if (response.status === 401) {
       const message = response.data.detail
       Message.error(message)
+    } else if (response.status === 400 && response.data.not_verified) {
+      Message.error({
+        duration: 0,
+        dangerouslyUseHTMLString: true,
+        message: 'Please verify your email before login in <i>Resend Email </i>'
```
Comment (michaelbukachi) [inline src/utils/request.js:53]: Use `message: this.$t('error.not_verified')`

Comment (michaelbukachi) [inline src/utils/request.js:53]: Use `message: Vue.i18n.translate('error.not_verified.description')`

Comment (lupyana) [reply src/utils/request.js:53]: I have tried this, I could not get it to work @michaelbukachi

File: src/lang/en.js [hunk 2]
```diff
@@ -54,6 +54,9 @@ export default {
       title: '403!',
       description: 'This action is forbidden',
       action: 'GO HOME'
+    },
+    not_verified: {
+      description: 'Please verify your email before login in'
```
Comment (andrewtpham) [inline src/lang/en.js:59]: Kindly change text to: `Please verify your email before logging in`

File: src/utils/request.js (hunk 1 above)
Comment (michaelbukachi) [reply src/utils/request.js:53]: Add this import at the top:
```
import i18n from '@/lang'
```
Then you can use it like so:
```
message: i18n.t('error.not_verified.description')
```

Comment (lupyana) [reply src/utils/request.js]: This solved it

File: src/utils/request.js [hunk 3]
```diff
@@ -46,6 +47,12 @@ service.interceptors.response.use(
     } else if (response.status === 401) {
       const message = response.data.detail
       Message.error(message)
+    } else if (response.status === 400 && response.data.not_verified) {
+      Message.error({
+        duration: 0,
+        dangerouslyUseHTMLString: true,
+        message: `${i18n.t('error.not_verified.description')}<i>Resend Email </i>`
```
Comment (michaelbukachi) [inline src/utils/request.js:54]: Don't you think that `Resend email` also needs to be translated? :smile:

**Comment by lupyana** [comment]
Yes, It can 

> hey @lupyana this looks really great! 
> 
> 1. Is it possible to change the style of the banner? I think if you use `h-alert` tag it should work
>    **Current**
>    image
> 
> **Expected**
> image
> 
> 1. Is it possible to make the 'Resend email' a clickable link. So that when users click on it a confirmation email can be sent? Let me know if we can keep it in this PR or make a separate ticket for this.
>    image

On it

**Comment by ninetteadhikari** [comment]
@lupyana let me know if the requested changes can be added to this PR or I should create a separate issue for it

**Comment by lupyana** [comment]
> @lupyana let me know if the requested changes can be added to this PR or I should create a separate issue for it

Yes it can be done

**Comment by lupyana** [comment]
@ninetteadhikari  @andrewtpham  I need to refactor the h-alert component to accept description as slot, and h-button of type text to remove shadow

**Comment by ninetteadhikari** [comment]
> refactor the h-alert component to accept more props i.e center and custom decription as slot

Hey @lupyana thanks that a good catch :) I created a ticket refactor the `h-alert` banner and assigned you if thats ok 🙂 https://github.com/hikaya-io/hakawati/issues/139

And for `h-button` would it be easier to just use `<a>` tags instead of buttons or do you think its better to refactor the buttons? let me know then i create relevant tickets for you!

**Comment by lupyana** [comment]
I will switch button for link

**Pull request merged**